openwisp_controller/connection/api/views.py

```diff
@@ -32,18 +29,11 @@ class ListViewPagination(pagination.PageNumberPagination):
     max_page_size = 100
 
 
-class BaseCommandView(GenericAPIView):
+class BaseCommandView(ProtectedAPIMixin):
+    org_field = 'device__organization'
     model = Command
+    queryset = Command.objects.prefetch_related('device')
```

Comment:
I would try to take advantage of this chance to add also `FilterByParentManaged`, see https://github.com/openwisp/openwisp-users#checking-parent-objects, the code for this mixin is here: https://github.com/openwisp/openwisp-users/blob/670ec9349c27813c130c9421c7b5d7e079230f9e/openwisp_users/api/mixins.py#L92-L122

That should allow to remove some redundant logic here.